### ISSUE TITLE
Error in as.Date.default(unique(data_all$date)) when model fitting 

### ISSUE THREAD

**Issue by cltorrens** [USGS-R/streamMetabolizer#387]
### Brief problem description
I keep getting an error message that I can't figure out when I run the streamMetabolizer models. 
The only date-time info in my dataset is the solar.time, created in streamMetabolizer from a correctly-formatted POSIXct date. There are gaps in my timeseries data, but I had understood that the package could handle that. Is that incorrect?
I appreciate any suggestions on how to proceed - Thank you!

### What you saw on your computer
I formatted the data as required, didn't change the specs for this test run, then tried to fit the model with both this code: 
```r
mm.bayes.vg <- metab(bayes_specs, data = DOdata1) #unchanged specs
```
AND this code: 
```r
mm.bayes.vg <- metab_bayes(bayes_specs, data = DOdata1) #unchanged specs
```
And got the following error message each time: 

Error in as.Date.default(unique(data_all$date)) : 
  do not know how to convert 'unique(data_all$date)' to class “Date”
Timing stopped at: 0.956 0.059 1.031

### Session information
```r
devtools::session_info()
```
─ Session info ────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       macOS High Sierra 10.13.6   
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/Denver              
 date     2020-04-22                  

─ Packages ────────────────────────────────────────────────────────────────────────────
 package           * version  date       lib source                                   
 assertthat          0.2.1    2019-03-21 [2] CRAN (R 3.5.2)                           
 backports           1.1.6    2020-04-05 [2] CRAN (R 3.5.2)                           
 broom               0.5.6    2020-04-20 [2] CRAN (R 3.5.2)                           
 callr               3.4.3    2020-03-28 [2] CRAN (R 3.5.2)                           
 cellranger          1.1.0    2016-07-27 [2] CRAN (R 3.5.0)                           
 cli                 2.0.2    2020-02-28 [1] CRAN (R 3.5.2)                           
 colorspace          1.4-1    2019-03-18 [2] CRAN (R 3.5.2)                           
 crayon              1.3.4    2017-09-16 [2] CRAN (R 3.5.0)                           
 DBI                 1.1.0    2019-12-15 [2] CRAN (R 3.5.2)                           
 dbplyr              1.4.3    2020-04-19 [2] CRAN (R 3.5.2)                           
 desc                1.2.0    2018-05-01 [2] CRAN (R 3.5.0)                           
 deSolve             1.28     2020-03-08 [1] CRAN (R 3.5.2)                           
 devtools            2.3.0    2020-04-10 [1] CRAN (R 3.5.2)                           
 digest              0.6.25   2020-02-23 [1] CRAN (R 3.5.2)                           
 dplyr             * 0.8.5    2020-03-07 [1] CRAN (R 3.5.2)                           
 ellipsis            0.3.0    2019-09-20 [1] CRAN (R 3.5.2)                           
 fansi               0.4.1    2020-01-08 [1] CRAN (R 3.5.2)                           
 forcats           * 0.5.0    2020-03-01 [2] CRAN (R 3.5.2)                           
 fs                  1.4.1    2020-04-04 [2] CRAN (R 3.5.2)                           
 generics            0.0.2    2018-11-29 [1] CRAN (R 3.5.0)                           
 ggplot2           * 3.3.0    2020-03-05 [2] CRAN (R 3.5.2)                           
 glue                1.4.0    2020-04-03 [1] CRAN (R 3.5.1)                           
 gtable              0.3.0    2019-03-25 [2] CRAN (R 3.5.2)                           
 haven               2.2.0    2019-11-08 [2] CRAN (R 3.5.2)                           
 hms                 0.5.3    2020-01-08 [2] CRAN (R 3.5.2)                           
 httr                1.4.1    2019-08-05 [2] CRAN (R 3.5.2)                           
 jsonlite            1.6.1    2020-02-02 [2] CRAN (R 3.5.2)                           
 LakeMetabolizer     1.5.0    2016-06-23 [1] CRAN (R 3.5.0)                           
 lattice             0.20-41  2020-04-02 [2] CRAN (R 3.5.2)                           
 lazyeval            0.2.2    2019-03-15 [2] CRAN (R 3.5.2)                           
 lifecycle           0.2.0    2020-03-06 [1] CRAN (R 3.5.2)                           
 lubridate         * 1.7.8    2020-04-06 [1] CRAN (R 3.5.1)                           
 magrittr            1.5      2014-11-22 [2] CRAN (R 3.5.0)                           
 memoise             1.1.0    2017-04-21 [1] CRAN (R 3.5.0)                           
 modelr              0.1.6    2020-02-22 [2] CRAN (R 3.5.2)                           
 munsell             0.5.0    2018-06-12 [2] CRAN (R 3.5.0)                           
 nlme                3.1-147  2020-04-13 [2] CRAN (R 3.5.2)                           
 pillar              1.4.3    2019-12-20 [1] CRAN (R 3.5.2)                           
 pkgbuild            1.0.6    2019-10-09 [2] CRAN (R 3.5.2)                           
 pkgconfig           2.0.3    2019-09-22 [2] CRAN (R 3.5.2)                           
 pkgload             1.0.2    2018-10-29 [2] CRAN (R 3.5.0)                           
 plyr                1.8.6    2020-03-03 [1] CRAN (R 3.5.2)                           
 prettyunits         1.1.1    2020-01-24 [2] CRAN (R 3.5.2)                           
 processx            3.4.2    2020-02-09 [2] CRAN (R 3.5.2)                           
 ps                  1.3.2    2020-02-13 [2] CRAN (R 3.5.2)                           
 purrr             * 0.3.4    2020-04-17 [1] CRAN (R 3.5.1)                           
 R6                  2.4.1    2019-11-12 [2] CRAN (R 3.5.2)                           
 Rcpp                1.0.4.6  2020-04-09 [1] CRAN (R 3.5.1)                           
 readr             * 1.3.1    2018-12-21 [2] CRAN (R 3.5.0)                           
 readxl              1.3.1    2019-03-13 [2] CRAN (R 3.5.2)                           
 remotes             2.1.1    2020-02-15 [2] CRAN (R 3.5.2)                           
 reprex              0.3.0    2019-05-16 [2] CRAN (R 3.5.2)                           
 rLakeAnalyzer       1.11.4.1 2019-06-09 [1] CRAN (R 3.5.2)                           
 rlang               0.4.5    2020-03-01 [1] CRAN (R 3.5.2)                           
 rprojroot           1.3-2    2018-01-03 [2] CRAN (R 3.5.0)                           
 rstudioapi          0.11     2020-02-07 [2] CRAN (R 3.5.2)                           
 rvest               0.3.5    2019-11-08 [2] CRAN (R 3.5.2)                           
 scales              1.1.0    2019-11-18 [2] CRAN (R 3.5.2)                           
 sessioninfo         1.1.1    2018-11-05 [2] CRAN (R 3.5.0)                           
 streamMetabolizer * 0.11.4   2020-04-22 [1] Github (USGS-R/streamMetabolizer@4a7ba9c)
 stringi             1.4.6    2020-02-17 [1] CRAN (R 3.5.2)                           
 stringr           * 1.4.0    2019-02-10 [2] CRAN (R 3.5.2)                           
 testthat            2.3.2    2020-03-02 [2] CRAN (R 3.5.2)                           
 tibble            * 3.0.1    2020-04-20 [1] CRAN (R 3.5.1)                           
 tidyr             * 1.0.2    2020-01-24 [1] CRAN (R 3.5.2)                           
 tidyselect          1.0.0    2020-01-27 [1] CRAN (R 3.5.2)                           
 tidyverse         * 1.3.0    2019-11-21 [2] CRAN (R 3.5.2)                           
 unitted             0.2.9    2018-05-02 [1] local                                    
 usethis             1.6.0    2020-04-09 [2] CRAN (R 3.5.2)                           
 utf8                1.1.4    2018-05-24 [2] CRAN (R 3.5.0)                           
 vctrs               0.2.4    2020-03-10 [1] CRAN (R 3.5.2)                           
 withr               2.2.0    2020-04-20 [2] CRAN (R 3.5.2)                           
 xml2                1.3.1    2020-04-09 [2] CRAN (R 3.5.2)                           
 yaml                2.2.1    2020-02-01 [2] CRAN (R 3.5.2)                           

[1] /Users/torrensc/Library/R/3.5/library
[2] /Library/Frameworks/R.framework/Versions/3.5/Resources/library

**Comment by aappling-usgs** [comment]
How are date gaps in your data represented? Do you still give them a row with some value for the solar.time such as NA, "NA", or FALSE? If so, try omitting those rows entirely.

If further debugging is needed, you could also try calling `as.Date(unique(DOdata1$solar.time))` - does that give you the same error?

**Comment by kqasem** [comment]
Hey,
I am having the same error. I have used the data before on another computer. When I installed R and the required packages on a different PC, I cant  "fit" the model on the same dataset.

Any luck?

**Comment by aappling-usgs** [comment]
Did you get my other message on Friday about day_start and day_end?

That’s useful to know that these same data run differently on different computers, thanks. And I’m glad you got through the RTools issue.

--
Alison Appling
U.S. Geological Survey
Data Science<https://www.usgs.gov/mission-areas/water-resources/science/data-science-water-resources> Branch, Integrated Information Dissemination Division

**Comment by aappling-usgs** [comment]
I have been able to reproduce this error on my computer with @kqasem's data and select choices of `day_start` and `day_end` in the specs. Some values work; others don't. I suggested that @kqasem use the defaults of 4 and 28, but @cltorrens, you say you're using the default specs, so you might get better results with 3 and 27, 4.5 and 28.5, or some other 24-hour period.

If either of you is able to provide additional information about which values you find that work or don't work, please let me know - the source of this error still mystifies me, so the more data the better.

**Comment by kqasem** [comment]
Alison,
I would like to thank you very much. Your emails were very helpful and I was able to read the data/run the model just fine. 
The issues has to do with the Rtools. I had to install the most recent "Frozen" version (35). That been said, I had to uninstall the current R version and install an older version that is compatible with Rtools35. After I did that, I was able to read my old data and even newer data.

Thanks again!
Karoline

**Comment by aappling-usgs** [comment]
Glad you were able to get things running, Karoline, and thanks for following up here! I'll keep an eye on the Rtools issue to see if I can discover a way to make streamMetabolizer compatible with the newest version; in the meantime, your solution will likely be useful to others.

**Comment by Jakob10er** [comment]
Hello,

I have the same Probelm. The model wont't work and i get the  error:

Error in as.Date.default(unique(data_all$date)) : 
  do not know how to convert 'unique(data_all$date)' to class “Date”

 When i try to use the model with the example data "data.metab()" it works, but not with my data.
Thee data frame I use is structered like this:

df <-  read.csv(paste0(westerhuesen2018,"metab_dataframe.csv"))

dataframe <- unitted(data.frame(solar.time = u(as.POSIXct(df$sol_time, tz = "UTC")),
                               DO.obs = u(df$ox.conc, "mgO2 L^-1"),
                               DO.sat = u(df$DO.sat, "mgO2 L^-1"),
                               depth = u(df$depth,"m"),
                               temp.water = u(df$Temp, "degC"),
                               light = u(df$light, "umol m^-2 s^-1"),
                               discharge = u(df$Abfluss, "m^3 s^-1 ")))

Are there any other ways known to solve the Probem with this error? Is there maybe a mistake when building the dataframe,  or am I doing an actually simple step wrong?
The solar time should be right. I used "convert_UTC_to_soalrtime()" for it

I already tried the things in the comments above (delete NA's, change day_start/day_end, use Rtools version 3.5)


Best regards
Jakob
